controllers/hyperconverged/hyperconverged_controller.go

```diff
@@ -481,41 +488,45 @@ func (r *ReconcileHyperConverged) handleUpgrade(req *common.HcoRequest, init boo
 	return nil, nil
 }
 
-func (r *ReconcileHyperConverged) handleNMO(req *common.HcoRequest) error {
-	oldNmoCRDobj := &apiextensionsv1.CustomResourceDefinition{}
-	oldNmoCRDkey := client.ObjectKey{
-		Namespace: "",
-		Name:      oldNmoCrdName,
-	}
-	err := r.client.Get(req.Ctx, oldNmoCRDkey, oldNmoCRDobj)
-	if err != nil {
-		if apimachineryerrors.IsNotFound(err) {
-			return nil
-		} else {
-			return err
+func (r *ReconcileHyperConverged) checkNMO(req *common.HcoRequest) error {
+	if hcoutil.GetClusterInfo().IsOpenshift() && hcoutil.GetClusterInfo().IsRunningOnOpenshift411OrLater(r.client) {
+		nmoCrdObj := &apiextensionsv1.CustomResourceDefinition{}
+		nmoCrdKey := client.ObjectKey{
+			Namespace: "",
+			Name:      nmoCrdName,
 		}
-	}
 
-	oldNmoCRs := &nmoapioldv1beta1.NodeMaintenanceList{}
-	if err := r.client.List(req.Ctx, oldNmoCRs); err != nil {
-		return err
-	}
-
-	for _, oldNmoCr := range oldNmoCRs.Items {
-		newNmoCr := &nmoapiv1beta1.NodeMaintenance{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: oldNmoCr.Name,
-			},
-			Spec: nmoapiv1beta1.NodeMaintenanceSpec{
-				NodeName: oldNmoCr.Spec.NodeName,
-				Reason:   oldNmoCr.Spec.Reason,
-			},
+		err := r.client.Get(req.Ctx, nmoCrdKey, nmoCrdObj)
+		if err != nil {
+			if apimachineryerrors.IsNotFound(err) {
+				return nil
+			} else {
+				return err
+			}
 		}
-		err := r.client.Create(req.Ctx, newNmoCr)
-		if err != nil && !apimachineryerrors.IsAlreadyExists(err) {
+
+		nmoCRs := &nmoapiv1beta1.NodeMaintenanceList{}
+		if err := r.client.List(req.Ctx, nmoCRs); err != nil {
 			return err
 		}
-		log.Info(fmt.Sprintf("Old Custom Resource Kind: %s Name: %s was converted to the new API", oldNmoCr.Kind, oldNmoCr.Name))
+
+		if len(nmoCRs.Items) > 0 {
+			req.Conditions.SetStatusCondition(metav1.Condition{
+				Type:               hcov1beta1.ConditionUpgradeable,
+				Status:             metav1.ConditionFalse,
+				Reason:             nmoCrExistErrorReason,
+				Message:            nmoCrExistErrorMessage,
+				ObservedGeneration: req.Instance.Generation,
+			})
+			r.eventEmitter.EmitEvent(req.Instance, corev1.EventTypeWarning, "NotUpgradeable",
+				fmt.Sprintf("Node Maintenance Operator custom resources %s have been found. Please remove them to allow upgrade. "+
+					"You can use NMO standalone operator if keeping the node(s) under maintenance is still required.", nmoCrdName))
+			req.Upgradeable = false
+			err = r.setOperatorUpgradeableStatus(req)
```

Comment:
Not sure it should be called from here. The `Reconcile` method should handle this call.